swath_projector/utilities.py

```diff
@@ -216,3 +228,34 @@ def make_array_two_dimensional(one_dimensional_array: np.ndarray) -> np.ndarray:
 
     """
     return np.expand_dims(one_dimensional_array, 1)
+
+
+def transpose_if_xdim_less_than_ydim(
+    variable_values: np.ma.MaskedArray,
+) -> np.ma.MaskedArray:
+    """Return transposed variable when variable is wider than tall.
+
+    QuickFix (DAS-2216): We presume that a swath has more rows than columns and
+    if that's not the case we transpose it so that it does.
+    """
+    if len(variable_values.shape) != 2:
+        raise ValueError(
+            f'Input variable must be 2 dimensional, but got {len(variable_values.shape)} dimensions.'
+        )
+    if variable_values.shape[0] < variable_values.shape[1]:
+        return np.ma.transpose(variable_values).copy()
+
+    return variable_values
+
+
+def get_rows_per_scan(total_rows: int) -> int:
+    """
+    Finds the smallest divisor of the total number of rows. If no divisor is
+    found, return the total number of rows.
+    """
+    if total_rows < 2:
+        return 1
```

Comment:
A question I have here is whether returning 1 is the right thing to do when the number of rows per scan is less than 2. Won't the EWA algorithm fail if we ask it to use less than 2 rows? If so, perhaps a better approach is to raise an exception here saying there aren't enough rows. The block of code that calls `pyresample.ewa.fornav` will raise an exception already, just potentially a less informative one.

(I'm not going to hold the PR up over this as there's already an exception being raised later on, but felt it was worth thinking a bit about - my thinking is that if we _know_ a return value of < 2 is problematic, then we should catch the problem as soon as we know about it)